packages/dcos-integration-test/extra/test_dcos_log.py

```diff
@@ -71,5 +72,145 @@ def test_log_proxy(cluster):
     for slave_id in slaves_ids:
         response = cluster.get('/system/v1/agent/{}/logs/v1/range/?skip_prev=10'.format(slave_id))
         check_response_ok(response, {'Content-Type': 'text/plain'})
-        lines = filter(lambda x: x != '', response.content.decode().split('\n'))
+        lines = filter(lambda x: x != '', response.content.decode('utf-8', 'ignore').split('\n'))
         assert len(list(lines)) == 10, 'Expect 10 log entries. Got {}. All lines {}'.format(len(lines), lines)
+
+
+def test_task_logs(cluster):
+    test_uuid = uuid.uuid4().hex
+
+    task_id = "integration-test-task-logs-{}".format(test_uuid)
+
+    task_definition = {
+        "id": "/{}".format(task_id),
+        "cpus": 0.1,
+        "instances": 1,
+        "mem": 128,
+        "cmd": "echo STDOUT_LOG; echo STDERR_LOG >&2;sleep 999"
+    }
+
+    with cluster.marathon.deploy_and_cleanup(task_definition, check_health=False):
+        url = get_task_url(cluster, task_id)
+        task_stdout_response = cluster.get('{}?filter=STREAM:STDOUT'.format(url))
+        check_response_ok(task_stdout_response, {})
+        task_stdout = task_stdout_response.content.decode('utf-8', 'ignore')
+        assert 'STDOUT_LOG' in task_stdout, 'Missing `STDOUT_LOG` in response. Got {}'.format(task_stdout)
+
+        task_stderr_response = cluster.get('{}?filter=STREAM:STDERR'.format(url))
+        check_response_ok(task_stderr_response, {})
+        task_stderr = task_stderr_response.content.decode('utf-8', 'ignore')
+        assert 'STDERR_LOG' in task_stderr, 'Missing `STDERR_LOG` in response. Got {}'.format(task_stderr)
+
+
+def get_task_url(cluster, task_name, stream=False):
```

Comment:
Would you prefer moving this function up, above line 78? ( The style varies here. but I have found dependencies (function definition) coming before invocation easier to navigate).